### ISSUE TITLE
New rule proposal: no-class

### ISSUE THREAD

**Issue by emmenko** [eslint/eslint#2097]
The aim of the rule is to forbid the use of `class` (ES6) to encourage other ways to do inheritance.

**Comment by michaelficarra** [comment]
Is this commonly desired? If so, what is the reasoning?

**Comment by nzakas** [comment]
This seems like a very personal constraint at the moment. Unless you can explain why this is common, I'd recommend creating a plugin for your custom rule.

**Comment by emmenko** [comment]
Well, I've read tons of articles that discourage the use of `class`(e.g.: https://medium.com/@_ericelliott/how-to-fix-the-es6-class-keyword-2d42bb3f4caf) and since I also wanted to avoid using it in my current project I thought to simply create a rule.
I don't actually see a problem (the rule is off by default and very minimalistic) and I don't consider it strictly personal. 
But if you guys think it should not be part of this library, or you don't consider it as a useful case no problem. 

@nzakas maybe can you explain how can I create a plugin?

**Comment by nzakas** [comment]
Why not just use `ecmaFeatures.classes` set to `false`? http://eslint.org/docs/configuring/#specifying-language-options

What more do you need?

**Comment by emmenko** [comment]
Tried that, didn't work. See https://github.com/eslint/eslint/pull/2096#issuecomment-83035980 please

**Comment by emmenko** [comment]
I've done some debugging and if I don't use the `babel-eslint` parser I correctly get the error

```bash
test-class.js
  1:1  error  Unexpected reserved word
```

Will file an issue there then.

**Comment by emmenko** [comment]
I've filed an issue https://github.com/babel/babel-eslint/issues/53

**Comment by emmenko** [comment]
So it won't work using `babel-eslint`. I'm creating a plugin then and use that 
https://github.com/emmenko/eslint-plugin-no-class
